packages/native-stack/src/types.tsx

```diff
@@ -558,10 +588,15 @@ export type NativeStackNavigationOptions = {
    * If set to non-negative value it will try to render sheet with provided radius, else it will apply system default.
    *
    * If left unset system default is used.
-   *
-   * @platform ios
    */
   sheetCornerRadius?: number;
+  /**
+   * Index of the detent the sheet should expand to after being opened.
+   * Works only when `stackPresentation` is set to `formSheet`.
+   *
+   * Defaults to `0` - which represents first detent in the detents array.
+   */
+  sheetInitialDetent?: number;
```

Comment:
I think `sheetInitialDetentIndex` would make more sense as you're not specifying the detent, but index of the detent.

```suggestion
  sheetInitialDetentIndex?: number;
```

Also what happens when you specify an index which is out of bounds? Does it throw an error? If not then it should.

Reply:
As to the naming? Won't it leave out the `sheetLargestUndimmedDetent`? It is also primarily an index (it has also additional options).

Reply:
I've aligned the naming. Both are now named `sheetInitialDetentIndex` and `sheetLargestUndimmedDetentIndex` accordingly. They both now accept `last` option instead of `largest`.